src/Core/Addons/Model/AddonManager.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Friendica\Core\Addons\Model;
+
+use Friendica\Core\Addons\Capabilities\IManageAddons;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\Hooks\Model\InstanceManager;
+use Friendica\Util\Strings;
+
+class AddonManager implements IManageAddons
+{
+	/** @var IManageConfigValues */
+	protected $config;
+	/** @var array */
+	protected $hooks;
+
+	public function __construct(IManageConfigValues $config)
+	{
+		$this->config = $config;
+	}
+
+	public function load(): self
+	{
+		$hooks = [];
+
+		$addons = array_filter($this->config->get('addons') ?? []);
+
+		foreach ($addons as $name => $data) {
+			$addonName     = Strings::sanitizeFilePathItem(trim($name));
+			$addonFilePath = sprintf('addon/%s/%s/%s', $addonName, static::ADDON_STATIC_DIR, static::ADDON_HOOKS_FILE);
+			if (!file_exists($addonFilePath)) {
+				continue;
```

Comment:
If the addon doesn’t register any hooks it can’t have Composer dependencies? 🤔 A positive structure (if file_exists then include) would have avoided this bug.